### ISSUE TITLE
Progress bar overflows, recovery is stuck.

### ISSUE THREAD

**Issue by nkrapivin** [francescotescari/XMT#35]
![image](https://user-images.githubusercontent.com/33228822/138418444-2b0bb0aa-fc9e-4c8a-904f-2f4ea96e92e5.png)

The recovery is stuck at 4.99%
What do I do??

Device: Redmi 10 selene

UPD: now it's at 52.49%, the UI is still overflowing.

UPD: It seems to have updated just fine, but it didn't wipe my settings (weird) and the UI was still bugged.

UPD: So yeah, it does update successfully even though the UI says that the installation had failed:

![image](https://user-images.githubusercontent.com/33228822/138423768-8a88d07a-3b58-4a50-b64a-f8fcbe4d813d.png)

**Comment by rasos** [comment]
Well, the percentage indicator on the device is jumping, yes. 

I am getting the same message that the sideload failed. Why do you think it was successful? My device Redmit Note 9 Pro is still not booting beyond the MIUI logo even when I try to install a stock ROM.

**Comment by modaali96** [comment]
did you fixed it ? 
i have the same problem

**Comment by Nik-Kot** [comment]
It is just an UI error. Everything flashed fin for me.

**Comment by Nik-Kot** [comment]
> Well, the percentage indicator on the device is jumping, yes.
> 
> I am getting the same message that the sideload failed. Why do you think it was successful? My device Redmit Note 9 Pro is still not booting beyond the MIUI logo even when I try to install a stock ROM.

Download a fastboot stock ROM and flash it using .bat (or .sh on Linux) file. **Important: do NOT choose the 'and lock' option**, if you do, you can potentially hard-brick your device.

**Comment by modaali96** [comment]
> 

Mine isn't working
The ui continues to about 500%
But the mobile stills 4.99% then it reboot and give me this error
And the phone still doesn't open
My phone can only open recovery and fastboot 
When I restart to system it closes 
It is not a boot loop it just closes faster one try 
Do you have faced the same problem?

**Comment by Nik-Kot** [comment]
> > 
> 
> Mine isn't working The ui continues to about 500% But the mobile stills 4.99% then it reboot and give me this error And the phone still doesn't open My phone can only open recovery and fastboot When I restart to system it closes It is not a boot loop it just closes faster one try Do you have faced the same problem?

No, I have not. Mine got stuck at 5% too, but after waiting a lot it continued, resulting is a working flash.

**Comment by Zonkil9** [comment]
I had the same today while updating Redmi Note 11 NFC (spesn). First, it stuck at 4.99%, then the progress bar in XiaoMiToolV2 overflown 100%, then it stuck at 52.49% on the phone, and, after a while, the phone updated successfully even though XiaoMiToolV2 showed it failed.

**Comment by LaurenceGough** [comment]
As said above just be patient and it will work. Great tool despite this nerve racking issue, all is good.

**Comment by th3fr3nchi3** [comment]
I went from 4.99% to 52.49%, it's just a scary visual bug, the flash went from EEA to Global.
